core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -1049,19 +1063,19 @@ private boolean isOverlapped(Pair<Long, Long> a, Pair<Long, Long> b) {
     @Override public void close() {
     }
 
-    private Object[] takeOne() {
-      return list.pollFirst();
+    private @Nullable Object[] takeOne() {
+      return requireNonNull(list.pollFirst(), "list.pollFirst()");
```

Comment:
This seems contradictive? `requireNonNull` indicates it should be non-nullable?

Reply:
This is just fine. `pollFirst` can return `null`, and here `non-null` is required by `takeOne`